dnf/base.py

```diff
@@ -2500,6 +2498,29 @@ def _report_already_installed(self, packages):
         for pkg in packages:
             _msg_installed(pkg)
 
+    def _raise_package_not_found_error(self, pkg_spec, forms, reponame):
+        all_query = self.sack.query(flags=hawkey.IGNORE_EXCLUDES)
+        subject = dnf.subject.Subject(pkg_spec)
+        solution = subject.get_best_solution(
+            self.sack, forms=forms, with_src=False, query=all_query)
+        if reponame is not None:
+            solution['query'].filterm(reponame=reponame)
+        if not solution['query']:
+            raise dnf.exceptions.PackageNotFoundError(_('No match for argument'), pkg_spec)
+        else:
+            with_modular_query = self.sack.query(flags=hawkey.IGNORE_MODULAR_EXCLUDES)
+            with_modular_query = solution['query'].intersection(with_modular_query)
+            with_regular_query = self.sack.query(flags=hawkey.IGNORE_REGULAR_EXCLUDES)
+            with_regular_query = solution['query'].intersection(with_regular_query)
+            if with_regular_query:
+                msg = _('All matches were excluded by regular filtering for argument')
+            elif with_modular_query:
+                msg = _('All matches were excluded by modular filtering for argument')
+            else:
+                msg = _('Argument matched only with source RPMs')
```

Comment:
What exactly is this case? Excuse my ignorance here, but briefly looking into this now, it seems dnf does not actually install source packages at all, right? In that case this message isn't actually useful at all, is it? Anyway, if we keep this message, I see this case has no test in the ci-dnf-stack PR, could you add a test case for it?

Reply:
I will add the test case. This situation happen happen when you try to install foo but only foo.src is available and no package provides foo.

Reply:
But it's not like you could actually install `foo.src` with dnf? So why don't you instead filter out the source packages from the query here, because the source packages aren't in the set of packages that could be installed either?